lib/bulk_processor/s3_file.rb

```diff
@@ -25,11 +25,15 @@ def exists?
     #
     # @yields [File] a local copy of the remote file
     def open
-      with_temp_file do |local_file|
-        object = client.get_object(bucket: bucket, key: key)
-        local_file.write(object.body.read)
-        local_file.rewind
-        yield local_file
+      begin
+        with_temp_file do |local_file|
+          object = client.get_object(bucket: bucket, key: key)
+          local_file.write(object.body.read)
+          local_file.rewind
+          yield local_file
+        end
+      rescue Aws::S3::Errors => e
+        raise "Aws::S3::Errors: #{e}, KEY: #{key}, BUCKET: #{bucket}"
```

Comment:
is there any AWS related rescue on top of it, which might get affected?
or should we raise same s3 error just to be sure we are not messing any upstream recues?

Reply:
@mkhandelwl Yes, thanks for calling this out. We should raise the same error. I'll make this update.